lib/middleware/csp.js

```diff
@@ -24,21 +45,49 @@ function createMiddleware(sCspUrlParameterName, oConfig) {
 		defaultPolicyIsReportOnly = false,
 		defaultPolicy2 = null,
 		defaultPolicy2IsReportOnly = false,
-		definedPolicies = {}
+		definedPolicies = {},
+		serveCSPReports = false
 	} = oConfig;
 
+
+	/**
+	 * List of CSP Report entries
+	 */
+	const cspReportEntries = [];
+
 	return function csp(req, res, next) {
 		const oParsedURL = parseurl(req);
 
 		if (req.method === "POST" ) {
 			if (req.headers["content-type"] === "application/csp-report" &&
-				oParsedURL.pathname.endsWith("/dummy.csplog") ) {
-				// In report-only mode there must be a report-uri defined
-				// For now just ignore the violation. It will be logged in the browser anyway.
+				oParsedURL.pathname.endsWith("/.ui5/csp/report.csplog") ) {
+				// Write the violation into an array
+				// They can be retrieved via a request to '/.ui5/csp/csp-reports.json'
+				if (serveCSPReports && req.body) {
+					const reqBodyObject = typeof req.body === "string" ? JSON.parse(req.body) : req.body;
+					const cspReportObject = reqBodyObject["csp-report"];
+					if (cspReportObject) {
+						// extract the csp-report and add it to the cspReportEntries list
+						cspReportEntries.push(cspReportObject);
+					}
+				}
 				return;
 			}
 			next();
 			return;
+		} else if (serveCSPReports && req.method === "GET") {
+			// serve csp reports
+			if (req.headers["content-type"] === "application/json" &&
+				oParsedURL.pathname.endsWith("/.ui5/csp/csp-reports.json")) {
```

Comment:
Same here, should be an absolute URL
```suggestion
				oParsedURL.pathname === "/.ui5/csp/csp-reports.json") {
```

Reply:
done